.github/actions/run-query/action.yml

```diff
@@ -0,0 +1,69 @@
+name: 'Execute Spice REPl query'
+description: 'Executes a query using Spice REPl and returns query result via GH Action output parameter'
+inputs:
+  query:
+    description: 'Query to execute'
+    required: true
+  start-runtime: 
+    description: 'Whether to start Spice runtime before executing the query'
+    required: false
+    default: 'true'
+  working-directory:
+    description: 'Directory to start Spice runtime'
+    required: false
+outputs:
+  query-result:
+    description: "Query result encoded as base64 string"
+    value: ${{ steps.query.outputs.query-result }}
+  runtime-trace:
+    description: "Spice runtime execution log"
+    value: ${{ steps.stop-runtime.outputs.runtime-traces }}
+runs:
+  using: "composite"
+  steps:
+    - name: Start Spice runtime
+      id: start-runtime
+      shell: bash
+      if: inputs.start-runtime == 'true'
+      working-directory: ${{ inputs.working-directory }}
+      run: |
+        spice run &> spice.log &
+        sleep 3
+        echo "Spice runtime started"
+    
+    - name: Run query
+      shell: bash
+      id: query
+      run: |
+        query="${{ inputs.query }}"
+        echo "Executing query: $query"
+        sql_output=$(spice sql <<EOF
+        $query
+        .error
+        EOF
+        )
+
+        echo "$sql_output"
+
+        if [[ $sql_output != *"No error to display"* ]]; then
+          echo "Query failed to execute."
+        fi
+
+        echo "query-output=$(echo "$sql_output" | base64 -w 0)" >> "$GITHUB_OUTPUT"
+
+        extracted_result=$(echo "$sql_output" | sed -n '/show tables;/,/Query took/p' | sed '1d;$d')
```

Comment:
This will need to take into account https://github.com/spiceai/spiceai/pull/1214